### DISCUSSION TITLE
I am at a loss

### DISCUSSION THREAD

**Discussion by acdbrn69** [ansible-collections/ansible.mysql#469] [Q&A]
So I am running a simple sql and always get an error 
  "msg": "A MySQL module is required: for Python 2.7 either PyMySQL, or MySQL-python, or for Python 3.X mysqlclient or PyMySQL. Consider setting ansible_python_interpreter to use the intended Python version."

I am running Python 3.x I checked to make sure I have the mods 

ubuntu@ip-10-245-160-87:~$ pip3 freeze | grep PyMySQL
PyMySQL==1.0.2
ubuntu@ip-10-245-160-87:~$ pip3 freeze | grep mysqlclient
mysqlclient==2.1.1

I spec the ansible_python_interpreter in my PB

---
- hosts: 10.245.xxx.xx
  name: Snipe-IT DB
  tags: t1
  gather_facts: yes
  become: true
  collections:
     - amazon.aws
     - community.aws
     - community.general
     - community.mysql.mysql_query

  tasks:

    - name: Simple select query to snipe db
      community.mysql.mysql_query:
       login_db: xxx
       login_user: xxxxx
       login_password: XXXXXXXX
       query: SELECT id FROM users
      vars:
       ansible_python_interpreter: /usr/bin/python3   

i spec'ed it out in the hosts file:

[automation_hosts:vars]
ansible_python_interpreter=/usr/bin/python3

even the command line:
ansible-playbook sql.yml -e 'ansible_python_interpreter=/usr/bin/python3' -vvv

all with no joy 

any ideas?

Thanks

W

**Comment by Andersson007** [comment]
@acdbrn69 hello, thanks for opening the discussion!
1. Can you re-check that the library is installed on the `target` machine where your DB lives (not on the `controller` one where ansible runs).
2. If you open a Python 3 interpreter on the target machine and try to run `import MySQLdb` will it work?
